### PR TITLE
Admin - Various Fixes

### PR DESCRIPTION
**Pull request by yj-soh**
Should be easier to review commit by commit.
- Allow null / empty in API's validators wherever suitable.
- Fix video element not stopping on page change in admin panel.
- Update navigation bar (the showing of sections authorised) after modifying self as an admin.
- Add logo into nav. bar
- Log out admin when accessing unauthorised resource / while unauthenticated / when scope changes.
- Restrict auth. scopes for some API endpoints more specifically.

### CODE REVIEW

File: app_server/app/controllers/AdminController.js [hunk 1]
```diff
@@ -189,13 +189,28 @@ Class.updateAdmin = function(request, reply) {
...
+    return updateCache(request, account, function() {
+      if (request.auth.credentials.userId === admin.userId) {
+        request.cookieAuth.set('scope', admin.permissions);
+      }
+
+      if (particulars.password) {
+        return reply(admin);
+      } else {
+        return reply(Utility.clearUserProfile(admin));
+      }
```
Comment (ndt93) [inline app_server/app/controllers/AdminController.js]: You can remove the `else` clause here.

File: app_server/app/controllers/AdminController.js [hunk 2]
```diff
@@ -288,12 +305,21 @@ var encrypt = function(password) {
...
 
 var updateCache = function(request, account, callback) {
-  request.server.app.cache.set(account.userId, account, 0, function (err) {
-    if (err) {
-      logger.error(err);
+  request.server.app.cache.get(account.userId, function(err, cached) {
+    if (cached && cached.item) {
+      // merge with cached version, with new taking priority
+      Object.keys(cached.item)
+            .filter((k) => !account.hasOwnProperty(k))
```
Comment (ndt93) [inline app_server/app/controllers/AdminController.js:308]: I think  `Object.keys` already enumerate object's own properties only so you can choose to remove the `filter` function.

**Comment by ndt93** [comment]
Some changes requested. Can merge after that.

File: app_server/app/controllers/AdminController.js (hunk 2 above)
Comment (yj-soh) [inline app_server/app/controllers/AdminController.js:308]: ? It's filtering out matching properties (keys) from `account`, though?
Like...

```
cached.item {'a': 1, 'b': 1, 'c': 1}
account {'a': 1, 'b': 1}
filtered ['c']
```

This segment is merging the `cached.item` and `account` objects.

Did I misunderstand something?

Comment (ndt93) [inline app_server/app/controllers/AdminController.js:308]: Ah ok sorry, I thought you were filtering out own properties of `cached`. That's good then.

**Pull request merged**